docs: UX strategy summary (#11)
Alzymologist/Kampela: docs/development/UX/strategy.md

# UX strategy

## Comfort of use

> The device should be pleasant to use. Users would have to hold it together with phone and flip this stack over, probably several times. We should not add unnecessary difficulties on top of that. Things like metadata management should happen automatically and without users' help.

Reviewer: ```suggestion
The device should be pleasant to use. Users would have to hold the phone and the Kampela as a stack and operate on it simultaneously. We should not add unnecessary difficulties on top of that. Things like metadata management should happen automatically and without users' help. 

Stack configuration must be researched and determined through physical. Some Possible options are:
- slide: Kampela screen is visible either under or over the phone screen
- book: Kampela screen is to the side of the phone screen

See [Kampela + phone positions](https://user-images.githubusercontent.com/69721131/194913904-f66504fe-1e86-4037-9076-773b9e0b05d9.jpeg)
```

Reviewer 2: It is helpful to have a picture here. I show some possibilities on the photo, but I'm still not sure what you meant by "flip". I would suggest to work around [book, slide-0, slide-1] positions since they are almost interchangeable with each other (they can be used with the same text direction on the screen), and also slide-1 seems the most natural.  

Should the text direction be vertical or horizontal? 
For better wrapping, it may be preferable for text to flow along the largest  dimension of the Kampela screen. It's better to test with the physical device.

![Kampela + phone positions](https://user-images.githubusercontent.com/69721131/194913904-f66504fe-1e86-4037-9076-773b9e0b05d9.jpeg)

Author: Yes, we'll have to study these UI questions later, this is just a strategy, where basis for future decision-making is laid out and questions (in addition to those already listed in security section) are raised.

We should probably mention somewhere here that these things should and will be answered through physical user testing. And photos, certainly, should be here.

Reviewer 3: Form factor
I think that the implementation of the device in the form of a phone case, for example, in the form of a book, will make it possible to justify the organization of the production of the device on a pre-order basis using 3D printing technology in manufacturing.

Author: Re-designing the casing means that we should re-do security analysis every time. We should separate the base device from extra convenience casing to be able to do this. Anyway, let's not put this into strategy, as this is not and could not be decided now, before mockup tests.

## Friction

> Similar to approach taken in Parity Signer development, certain things that are allowed but only should be done with care should result in friction. All transactions must be read and explicitly confirmed before the secret storage access. Removal/change of trust sources should be sufficiently difficult to stop users from doing it mindlessly.

Reviewer: ```suggestion
Similar to approach taken in Parity Signer development, certain allowed actions that require extra attention should result in friction. All transactions must be read and explicitly confirmed before accessing the secret storage. Removal/change of trust sources should be sufficiently difficult to stop users from doing it mindlessly.
```

## Forbidden actions

Certain actions should be forbidden altogether even if complete device replacement is the only alternative.

## Disposable Kampela

Users should be ready to get a new Kampela if something goes wrong. There should be a simple and clear safe disposal protocol that anyone could follow - even without authorization. Safe disposal is essentially the best scenario among DOS attacks (which are quite trivial with physical access to device) and should be encouraged.

## Personalization

> Kampelas should be highly personizeable, probably in subtle ways. It should be trivial to observe that the device was swapped.

Reviewer: ```suggestion
Kampelas should be highly personizable, probably in subtle ways. It should be trivial for owner to observe that the device was swapped.
```

## Few buttons

Due to power and computational limitation, on-board controls would be inevitably limited. Thus the buttons should be really good, reusable and easy to understand.

## Device versions

> Users would not have access to device firmware (except for development model probably). Some options that could only be set once (post-setup seed backup, verifier change, derivations enablement) could be factory-set so that there are just many different models of Kampela available for customers.

Reviewer: ```suggestion
Users will not have access to device firmware (except for development model probably). Users may want to select own settings during the Kampela initialization (allow secret backup, allow verifier to be changed, block more than one derivation per device). These could be also factory-set so that there are just many different models of Kampela available for customers.
```

## Signer ecosystem

Kampela should fit nicely within Parity Signer ecosystem and - where possible - should adapt UI components from there.

## Utilize first use moment

> The user would be really vigilant only once - on first use. Thus we should utilize this unique moment, minimize amount of annoying attentios-diluting things and maximize educational impact.

Reviewer: ```suggestion
The user will be really vigilant only once - on first use. Thus we should utilize this unique moment, minimize amount of annoying attention diluting things and maximize educational impact.
```

> 

Reviewer: ```suggestion
## Manuals

Kampela must be accompanied with sufficient instructions and manuals, possibly even in printed form.
```

## Personalization

Kampelas should be highly personizable, probably in subtle ways. It should be trivial for owner to observe that the device was swapped.

## Few buttons

Due to power and computational limitation, on-board controls would be inevitably limited. Thus the buttons should be really good, reusable and easy to understand.

## Device versions

Users will not have access to device firmware (except for development model probably). Users may want to select own settings during the Kampela initialization (allow secret backup, allow verifier to be changed, block more than one derivation per device). These could be also factory-set so that there are just many different models of Kampela available for customers.

## Signer ecosystem

Kampela should fit nicely within Parity Signer ecosystem and - where possible - should adapt UI components from there.

## Utilize first use moment

The user will be really vigilant only once - on first use. Thus we should utilize this unique moment, minimize amount of annoying attention diluting things and maximize educational impact.

> ## Manuals

Reviewer: ```suggestion

## Manuals
```

# UX strategy

## Comfort of use

The device should be pleasant to use. Users would have to hold the phone and the Kampela as a stack and operate on it simultaneously. We should not add unnecessary difficulties on top of that. Things like metadata management should happen automatically and without users' help.

Stack configuration must be researched and determined. Possible options are:
- flip: Kampela screen is on the opposite side of the phone screen
- slide: Kampela screen is visible either under or over the phone screen
- book: Kampela screen is to the side of the phone screen

> 

Author: ```suggestion

UI decisions on device shape and handling should be made based on live user tests with device mock models or prototypes.

```

> # UX strategy

Author: ```suggestion
# UX overview

The Kampela system consists of Kampela itself and optionally removable NFC token independent from the device but required for authorization and operation of the tool.

For proper signing operation, a hot wallet is required for input and output. In general, these could be two different wallets, the only requirement is that these are brought into network connection within reasonable time before and after transaction signing. In best case scenario there is just one duplex hot wallet.

An optional encoding device could be used if the input hot wallet does not support NFC interfacing.

Thus, we have 3-6 devices that should stay in close proximity of each other, within close distance from external network for Substrate access and user, that handles these devices.

# UX strategy
```

> 

Reviewer 3: I think we are missing the fact that we have 3 devices: HW - K - SMF
And it is quite clear that they cannot be far from each other.
It follows that the conditional UX user cannot be anywhere, but only where it is convenient to perform the necessary procedures and only where there is a stable connection. Then we can imagine a certain conditional scheme of the functioning of the overall system:

Author: See the terminology in suggestion. Let's not label anything but Hot Wallet with HW, or get rid of these abbreviation altogether.

> 

Reviewer 3: In addition, I propose to add a data transfer function to K when replacing the wallet itself. I think that such an opportunity can increase the rating of the device.

Author: see suggestion below

## Comfort of use

The device should be pleasant to use. Users would have to hold the phone and the Kampela as a stack and operate on it simultaneously. We should not add unnecessary difficulties on top of that. Things like metadata management should happen automatically and without users' help.

Stack configuration must be researched and determined. Possible options are:
- flip: Kampela screen is on the opposite side of the phone screen
- slide: Kampela screen is visible either under or over the phone screen
- book: Kampela screen is to the side of the phone screen

UI decisions on device shape and handling should be made based on live user tests with device mock models or prototypes.

## Friction

Similar to approach taken in Parity Signer development, certain allowed actions that require extra attention should result in friction. All transactions must be read and explicitly confirmed before accessing the secret storage. Removal/change of trust sources should be sufficiently difficult to stop users from doing it mindlessly.

## Forbidden actions

Certain actions should be forbidden altogether even if complete device replacement is the only alternative.

> 

Reviewer 3: As for the replacement:
Here I see 2 options - in all three cases, the issue must be resolved without additional disassembly, otherwise it will be necessary to maintain an additional staff with appropriate logistics.
1 Replacement due to malfunction
2 Replacement due to discredit
In both cases, it is advisable to use the same protocol. The protocol should provide for remote registration in the system of the very fact of using the protocol and remote disposal (the technical possibilities of self-destruction of either the device or its contents should be considered, excluding the very possibility of using the contents of the device).

Author: The offline nature of the Kampela makes any remote actions impossible. We would have to rely on difficulty of secret extraction from Kampela and on user's vigilance to deal with possible breach in case the device was lost.

In all other scenarios user should safely dispose of device. We would have no tracking of the devices state, again, for privacy and due to our inability to effectively track air-gapped and powered-off for the most of the time Kampelas.

Here by "replacement" I mean that if user wants to perform forbidden actions, they might need to buy a new Kampela and destroy the old one.

## Disposable Kampela

Users should be ready to get a new Kampela if something goes wrong. There should be a simple and clear safe disposal protocol that anyone could follow - even without authorization. Safe disposal is essentially the best scenario among DOS attacks (which are quite trivial with physical access to device) and should be encouraged.

## Personalization

> Kampelas should be highly personizable, probably in subtle ways. It should be trivial for owner to observe that the device was swapped.

Reviewer 2: ```suggestion
Kampelas should be highly personalizable, probably in subtle ways. It should be trivial for owner to observe that the device was swapped.
```

## Few buttons

Due to power and computational limitation, on-board controls would be inevitably limited. Thus the buttons should be really good, reusable and easy to understand.

## Device versions

Users will not have access to device firmware (except for development model probably). Users may want to select own settings during the Kampela initialization (allow secret backup, allow verifier to be changed, block more than one derivation per device). These could be also factory-set so that there are just many different models of Kampela available for customers.

## Signer ecosystem

Kampela should fit nicely within Parity Signer ecosystem and - where possible - should adapt UI components from there.

## Utilize first use moment

The user will be really vigilant only once - on first use. Thus we should utilize this unique moment, minimize amount of annoying attention diluting things and maximize educational impact.

## Manuals

> Kampela must be accompanied with sufficient instructions and manuals, possibly even in printed form.

Author: ```suggestion
Kampela must be accompanied with sufficient instructions and manuals, possibly even in printed form.

## Migration

When migrating between devices, secret information should be entered anew, since there is no interdevice communication channel (and should not be). All non-secret information could be transferred through NFC from hot wallet; we need to establish the derivation export protocol akin to one used in Signer for batch derivation import, but with broader functionality to make migration as simple as possible (even though inevitably difficult due to seed phrase re-entry at least; see Friction section above).

```

# UX overview

The Kampela system consists of Kampela itself and optionally removable NFC token independent from the device but required for authorization and operation of the tool.

For proper signing operation, a hot wallet is required for input and output. In general, these could be two different wallets, the only requirement is that these are brought into network connection within reasonable time before and after transaction signing. In best case scenario there is just one duplex hot wallet.

An optional encoding device could be used if the input hot wallet does not support NFC interfacing.

> Thus, we have 3-6 devices that should stay in close proximity of each other, within close distance from external network for Substrate access and user, that handles these devices.

Reviewer 2: ```suggestion
Thus, we have from 3 (Kampela + NFC_token + Phone) to 6 devices that should stay in close proximity of each other, within close distance from external network for Substrate access and user, that handles these devices.
```
